src/Invoice/Calculator/FinancialYearCalculator.php

```diff
@@ -0,0 +1,114 @@
+<?php
+
+/*
+ * This file is part of the Kimai time-tracking app.
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace App\Invoice\Calculator;
+
+use App\Configuration\SystemConfiguration;
+
+class FinancialYearCalculator
+{
+    private \DateTimeInterface|null $financialYearStart = null;
+    private \DateTimeInterface|null $financialYearEnd = null;
+
+    public function __construct(private SystemConfiguration $systemConfiguration)
+    {
+        if (!$financialYearStart = $this->systemConfiguration->getFinancialYearStart()){
+            return;
+        }
+        $this->financialYearStart = \DateTime::createFromFormat('Y-m-d', $financialYearStart);
+        $this->setFinancialYearEndFromStartDate($this->financialYearStart);
+    }
+
+    private function setFinancialYearEndFromStartDate(\DateTimeInterface $financialYearStart): void
+    {
+        $this->financialYearEnd = $financialYearStart
```

Comment:
You are working with a DateTimeInterface and those need to be treated as Immutable.

Please change all ocurences of DateTime to DateTimeImmutable.